src/Protocol/DocumentUri.cs

```diff
@@ -14,17 +15,23 @@ namespace OmniSharp.Extensions.LanguageServer.Protocol
     /// <remarks>This exists because of some non-standard serialization in vscode around uris and .NET's behavior when deserializing those uris</remarks>
     public class DocumentUri : IEquatable<DocumentUri>
     {
+        public static Regex WindowsPath =
+            new Regex("^\\w(?:\\:|%3a)[\\\\|\\/]", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+        public string _delimiter = SchemeDelimiter;
+
         /// <summary>
         /// Create a new document uri
         /// </summary>
-        /// <param name="url"></param>
+        /// <param name="url"></param> add .
         public DocumentUri(string url)
         {
+            var uncMatch= false;
             var delimiterIndex = url.IndexOf(SchemeDelimiter, StringComparison.Ordinal);
-            if (delimiterIndex == -1)
+            if ((uncMatch = url.StartsWith("\\\\")) || (url.StartsWith("/")) || (WindowsPath.IsMatch(url)))
```

Comment:
I'm trying to think if it's safer to only do this on Windows... Since UNC style paths are not a thing on unix.

Reply:
It doesn't matter because later on we special case unc later on in the flow.  Basically that check is to handle the well known non-uri paths that we can convert into a file uri.